src/sensors/IMD.py

```diff
@@ -0,0 +1,74 @@
+from pin.pinout import Pinout
+from services.PWM import PWM
+from services.digital_in import DigitalInService
+from shared_memory import SharedMemory
+from test_lib.input.aggregate import Multiple
+from test_lib.input.input import Input
+from test_lib.input.timed import After
+from enum import Enum
+
+class Case(Enum):
+    ShortCircuit    = 0
+    Normal          = 10
+    UnderVoltage    = 20
+    EquipmentFault  = 40
+    GroundFault     = 50
+
+class IMDInput(Input):
+    def __init__(self, frequency : int, duty_cycle : int):
+        self.frequency = frequency
+        self.duty_cycle = duty_cycle
+
+    async def apply(self):
+        self.m_hs.set_frequency(self.frequency)
+        self.m_hs.set_duty_cycle(self.duty_cycle)
```

Comment:
you don't need to do this as the Input will be defined by @oganigl once his PR is merged, talk to him about the interface and assume it will be done at some point, but the PWM input should be the same as what you are doing here

Reply:
basically, this implementation is responsability of the PWM/InputCapture service, not the IMD